gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -28,10 +29,63 @@
 
 from gcp_variant_transforms.beam_io import vcfio
 
-__all__ = ['VcfHeader', 'VcfHeaderSource', 'ReadAllVcfHeaders',
-           'ReadVcfHeaders', 'WriteVcfHeaders']
+__all__ = ['HeaderConflictResolver', 'VcfHeader', 'VcfHeaderSource',
+           'ReadAllVcfHeaders', 'ReadVcfHeaders', 'WriteVcfHeaders']
 
 
+class HeaderConflictResolver(object):
+  """ A class for resolving mistmatch between header fields
+      (e.g. type, num).
+  """
+
+  # List of all resolvable conflicts with their resolution.
+  # For example, entry [a, b, c] means mistmatch between a and b is resolved to
+  # c (where c is usually either a or b).
+  _RESOLVABLE_CONFLICTS = [
+      # Conflict between 'Integer' and 'Float' is resolved to 'Float'.
+      ['Integer', 'Float', 'Float'],
+  ]
+
+  def __init__(self):
+    """ Initialize :class:`HeaderConflictResolver` by creating a dict of all
+        resolvable conflicts.
+    """
+    self._resolvable_conflicts_dict = {}
+    self._resolvable_conflicts_dict.update(
+        {rc[0] : {rc[1] : rc[2]} for rc in self._RESOLVABLE_CONFLICTS})
+    self._resolvable_conflicts_dict.update(
+        {rc[1] : {rc[0] : rc[2]} for rc in self._RESOLVABLE_CONFLICTS})
+
+  def can_resolve(self, first, second):
+    """ Returns true iff conflict between `first` and `second` is resolvable.
+
+    Order of args does not matter.
+
+    Args:
+      first_value (string): first given value.
+      second_value (string): second given value.
+    """
+    return (first == second or
+            (first in self._resolvable_conflicts_dict and
+             second in self._resolvable_conflicts_dict[first]))
+
+  def resolve(self, first, second):
+    """ Returns the resolution if the conflict between `first` and `second`
+    is resolvable.
+
+    Order of args does not matter.
+
+    Args:
+      first_value (string): first given value.
+      second_value (string): second given value.
+    """
+    if not self.can_resolve(first, second):
+      return None
```

Comment:
I think in this case we should raise the `'Incompatible number or types in header fields: ' {}, {}'.format(source_value, to_merge_value)` error since only resolvable values should be passed to the method.

Reply:
Done!